prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -0,0 +1,194 @@
+# DELETED: comm_type, LOAD, vmtsc
+
+import mileage
+import charging
+import numpy as np
+
+
+def immediate_charging(
+    census_region, model_year, veh_range, kwhmi, power, location_strategy
+):
+    """Immediate charging function
+
+    :param int census_region: any of the 9 census regions defined by US census bureau.
+    :param int model_year: year that is being modelled/projected to, 2017, 2030, 2040, 2050.
+    :param int veh_range: 100, 200, or 300, represents how far vehicle can travel on single charge.
+    :param int kwhmi: fuel efficiency, should vary based on vehicle type and model_year.
+    :param int power: charger power, EVSE kW.
+    :param int location_strategy: where the vehicle can charge-1, 2, or 3; 1 = home only, 2 = home and work, 3 = everywhere
+    :return: (*numpy.ndarray*) -- charging profiles.
+    """
+    # load NHTS data from function
+    newdata = mileage.remove_ldt(mileage.load_data(census_region))
+    # add new columns to newdata to store data that is not in NHTS data
+    newdata.insert(28, "trip start battery charge", [0] * len(newdata))
+    newdata.insert(29, "trip end battery charge", [0] * len(newdata))
+    newdata.insert(30, "charging power", [0] * len(newdata))
+    newdata.insert(31, "charging time", [0] * len(newdata))
+    # jx: never actually used... delete?
+    newdata.insert(32, "charging consumption", [0] * len(newdata))  # needs double check
+    newdata.insert(33, "BEV could be used", [0] * len(newdata))
+    newdata.insert(34, "trip number", [0] * len(newdata))
+
+    input_day = mileage.get_input_day(mileage.get_model_year_dti(model_year))
+
+    TRANS_charge = np.zeros(24 * len(input_day))
+
+    data_day = mileage.get_data_day(newdata)
+
+    actual_vmt = 0
+    potential_vmt = 0
+
+    # from grid to battery efficiency
+    charging_efficiency = 0.9
+
+    # charging when SOC less than batterstrategy
+    battery_strategy = 1
+
+    # 1-anytrip number, 2-last trip
+    trip_strategy = 1
+
+    kwh = kwhmi * veh_range
+
+    for day_iter in range(len(input_day)):
+
+        electricload = np.zeros(4800)

```

Comment:
What is the purpose of this number?

Reply:
It seems like maybe what we're doing is calculating electrical load over two days by the hundredth of an hour, before downsampling to hourly resolution. If so, we should be able to do this in a more automated way using the `resample` method of a time-indexed data frame: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.resample.html